src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4119,10 +4116,13 @@ private ImmutableArray<IPropertySymbol> GetIndexerGroupSemanticSymbols(BoundExpr
                 return ImmutableArray<IPropertySymbol>.Empty;
             }
 
-            return FilterOverriddenOrHiddenIndexers(symbols.ToImmutableAndFree());
+            var result = FilterOverriddenOrHiddenIndexers(symbols);
+            symbols.Free();
+
+            return result;
         }
 
-        private static ImmutableArray<IPropertySymbol> FilterOverriddenOrHiddenIndexers(ImmutableArray<ISymbol> symbols)
+        private static ImmutableArray<IPropertySymbol> FilterOverriddenOrHiddenIndexers(ArrayBuilder<ISymbol> symbols)
```

Comment:
>FilterOverriddenOrHiddenIndexers [](http://example.com/codeflow?start=55&length=32)

It looks like there is only one call site for this helper. Therefore, we could make it return void and simply remove filtered out items from the builder, if any.

Reply:
Input array is of ISymbol, while output array is of type IPropertySymbol.